hello-neon/app/src/main/cpp/CMakeLists.txt

```diff
@@ -12,21 +12,19 @@ add_library(cpufeatures  STATIC
 set(hello_neon_SRCS helloneon.c)
 
 # set up neon build flag for file using intrinsics
-# name: helloneon-intrinsics.c.neon  (It is named EXACTLY as this on disk)
-#                                    (you could name it otherwise)
-# 1) Inform CMake this file is part of the C source file
-# 2) Set up neon flag for neon file
+# name: helloneon-intrinsics.c (It is named EXACTLY as this on disk,
+#                              just like a normal source file)
+# then set up neon flag for neon files
 # [This example only build for armeabi-v7a, x86 could be done the same way]
 #
 if (${ANDROID_ABI} STREQUAL "armeabi-v7a")
   set(hello_neon_SRCS ${hello_neon_SRCS}
-                      helloneon-intrinsics.neon)
+                      helloneon-intrinsics.c)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DHAVE_NEON=1")
```

Comment:
HAVE_NEON need to be added to other files: if it is NOT neon, it will NOT call the neon implementation at all, so it is used there too.